### ISSUE TITLE
Hard to follow releases

### ISSUE THREAD

**Issue by jeffmaury** [eclipse/lsp4mp#141]
Looking at http://download.eclipse.org/lsp4mp/releases/ you can guess there is 0.2.1 release but can't find corresponding tag/commit in source tree
Also I would expect that after a release is pushed I could find a minor/micro in snapshots as, in JBoss Tools, after a GA is released (based on one LSP4MP release), we start the next cycle with a snapshot

**Comment by datho7561** [comment]
We need to make sure the tagged commit is attached to the tree (currently the commit that is tagged is detached from the tree).

We should also look into automating the version bump as part of the release pipeline through the GitHub bot. Doing it manually, it can get stuck in review: https://github.com/eclipse/lsp4mp/pull/139.

**Comment by rgrunber** [comment]
I alias `git tree` to `log --graph --decorate --oneline --abbrev-commit --all` using the aliases entry in my .gitconfig . This allows me to see all commits associated with a branch or tag. I'm not sure if you're just using a git log, but that would always be limited to commits/branches/tags associated with the current branch.

I agree though that the release commits being detached from any branch was a bit weird, but I think it makes sense in a historical sense. The commit is not part of development. The downside is any issues that come up with that commit become a bit tricky to resolve.

If we really want the tag to reside on the development branch, we just need to avoid being in a detatched (HEAD) state when the build pushes it (ie. checkout master).

**Comment by jeffmaury** [comment]
Yes I was confused by the GitHub UI so I can see the 0.2.1 tag but I think master/main branch should be updated to 0.2.2 as soon as 0.2.1 is released

**Comment by rgrunber** [comment]
We should be on 0.3.0-SNAPSHOT. Closing for now.
